### PR TITLE
contracts: Reorganize and hardhat tests

### PR DESCRIPTION
**Pull request by blukat29**
## Proposed changes

Reorganize contracts and add hardhat tests. See [contracts/README.md](https://github.com/blukat29/klaytn/blob/reorganize-contracts/contracts/README.md) for an overview.

```
cd contracts
npm install
npx hardhat test
```

## Types of changes

- [ ] Bugfix
- [ ] New feature or enhancement
- [x] Others

## Checklist

- [x] I have read the [CONTRIBUTING GUIDELINES](https://github.com/klaytn/klaytn/blob/master/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://cla-assistant.io/klaytn/klaytn)
- [ ] Lint and unit tests pass locally with my changes (`$ make test`)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments

Contracts reorganize
- Step 1. (This PR) #2124 - Add contracts in the new directory structure
- Step 2. (Following PR) #2127 - Remove old contract files, Fix import paths in Go codes

### CODE REVIEW

**Review by ian0371 — APPROVED** [review]

**Review by ian0371 — APPROVED** [review]

File: contracts/contracts/libs/openzeppelin-contracts-v2/contracts/drafts/.gitkeep
Comment (yoomee1313) [inline contracts/contracts/libs/openzeppelin-contracts-v2/contracts/drafts/.gitkeep:1]: `.gitkeep` seems no need since `drafts` folder is not empty.

Comment (blukat29) [reply contracts/contracts/libs/openzeppelin-contracts-v2/contracts/drafts/.gitkeep:1]: This file is from the original openzeppelin-contracts codebase so I want to refrain from modifying it.

Comment (yoomee1313) [reply contracts/contracts/libs/openzeppelin-contracts-v2/contracts/drafts/.gitkeep:1]: I didn't know that. Understood.

**Comment by hyunsooda** [comment]
I found some of duplicated files, for example

```
contracts/testing/sc_erc20/sc_token.sol
sc_erc20/sc_token.sol

contracts/service_chain/IERC20BridgeReceiver.sol
sc_erc20/IERC20BridgeReceiver.sol

...
```

What was the purpose of having identical files in directories at different levels?

**Comment by blukat29** [comment]
@hyunsooda The sc_erc20 directory (i.e. "old" contract files) will be removed by the Step 2 pull request.

**Review by hyunsooda — APPROVED** [review]

**Pull request merged**